### PR TITLE
Add episode 25

### PR DESCRIPTION
**Pull request by bitsondatadev**

### CODE REVIEW

File: _episodes/25.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ * Gunnar Morling, Lead of Debezium and Open source software engineer at [Red Hat](https://www.redhat.com)
+ ([@gunnarmorling](https://twitter.com/gunnarmorling)).
+ * Ashhar Hasan, Software Engineer at [Starburst](https://starburst.io/)
+ ([@hashhar](https://twitter.com/hashhar)).
+ 
+## Release 361
+
+Official announcement items from Martin:
+
+*
```
Comment (mosabua) [inline _episodes/25.md:32]:
```
* Support for OAuth2/OIDC opaque access tokens
* Aggregation pushdown for Pinot
* Better performance for Parquet files with column indexes
* Support for reading fields as JSON values in Elasticsearch
```

File: _episodes/25.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+ 
+## Release 361
+
+Official announcement items from Martin:
+
+* 
+
+Manfred's additional notes:
+
+*
```
Comment (mosabua) [inline _episodes/25.md:36]:
```
* Predicate pushdown in Cassandra
* Metadata cache size limitation in a few connectors
* Lots of improvements for Hive view support
* Glue table statistics improvements
```

File: _episodes/25.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
+---
+layout: episode
+title:  "25: Trino Going Through Changes"
```
Comment (mosabua) [inline _episodes/25.md:3]: Trino going through changes

or 

Trino going through the changelog

File: _episodes/25.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+     time: 2747
+   - title: "PR of the week"
+     desc: "PR 4140 Implement aggregation pushdown in Pinot"
+     time: 3865
+   - title: "Question of the week"
+     desc: "Is there an array function that flattens a row into three rows?"
+     time: 4265
+---
+
+## Guests
```
Comment (mosabua) [inline _episodes/25.md:26]: Empty line after section title

File: _episodes/25.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+If you know Trino, you know it allows for flexible architectures that include 
+many systems with varying use cases they support. We've come to accept this 
+potpourri of systems as a general modus operandi for most businesses. 
+
+Many times the data gets copied to different systems to accomplish varying use 
+cases from performance and data warehousing to merge cross cutting data into a 
+single store. When copying data between systems, how do these systems stay in 
+sync? It's a critical need especially for Trino to know that the state across 
+the data sources we query are valid.
```
Comment (mosabua) [inline _episodes/25.md:62]: is valid

as in 

the state is valid

File: _episodes/25.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+Many times the data gets copied to different systems to accomplish varying use 
+cases from performance and data warehousing to merge cross cutting data into a 
+single store. When copying data between systems, how do these systems stay in 
+sync? It's a critical need especially for Trino to know that the state across 
+the data sources we query are valid. 
+
+To answer this, we can use the concept of Change Data Capture (CDC). CDC is a 
+powerful concept that considers a data source(s), called a system of record(s), 
+that store the true state of a system. The system of records are monitored for
```
Comment (mosabua) [inline _episodes/25.md:66]: systems of record

File: _episodes/25.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+One implemention of CDC that has grown tremendously in popularity since its 
+inception is called Debezium. According to <https://debezium.io>:
+
+> Debezium is an open-source distributed platform for change data capture. Start
+> it up, point it at your databases, and your apps can start responding to all 
+> of the inserts, updates, and deletes that other apps commit to your databases.
+> Debezium is durable and fast, so your apps can respond quickly and never miss
+> an event, even when things go wrong.
+
+The common way Debezium is deployed in the wild is using Kafka Connect as the
```
Comment (mosabua) [inline _episodes/25.md:86]: make Kafka Connect a link

File: _episodes/25.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+CDC servers by defining the Debezium source connectors. You can then use the
```
Comment (mosabua) [inline _episodes/25.md:87]: Also .. reword this sentence to be clearer .. I think Kafka Connect is only used for targets .. not to pull data from source

File: _episodes/25.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+The common way Debezium is deployed in the wild is using Kafka Connect as the
+CDC servers by defining the Debezium source connectors. You can then use the 
+Kafka Connect ecosystem to sink to different targets downstream.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/25/debezium-architecture.png"/><br/>
+The Debezium architecture with Kafka Connect: <a href="https://debezium.io/documentation/reference/architecture.html">Source</a>.
+</p>
+
+Another alternative if you don't want to use Kafka, is to use dedicated Debezium
```
Comment (mosabua) [inline _episodes/25.md:95]: alternative, if

File: _episodes/25.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+servers to implement CDC and push the logs to target using Debezium connectors.
```
Comment (mosabua) [inline _episodes/25.md:96]: to target ??

File: _episodes/25.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/25/debezium-server-architecture.png"/><br/>
+The Debezium standalone server architecture: <a href="https://debezium.io/documentation/reference/architecture.html">Source</a>.
+</p>
+
+While CDC is the primary focus Debezium also provides support for more advanced
```
Comment (mosabua) [inline _episodes/25.md:103]: focus,

File: _episodes/25.md [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+concepts such as the [outbox pattern support for Quarkus apps](https://debezium.io/documentation/reference/integrations/outbox.html).
+
+### Debezium + Trino at Zomato
+
+Zomato is a technology platform that connects customers, restaurant partners and
```
Comment (mosabua) [inline _episodes/25.md:108]: This kind of info should also be talked about in the actual event..

File: _episodes/25.md [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+Here is one of the earlier iterations of Zomato's big data architecture before
+they were able to integrate Debezium. Ayush covers some of the pain points they
+experienced before implementing CDC.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/25/zomato-before.png"/><br/>
+</p>
+
+Once Zomato implemented CDC, they were able to keep their downstream Iceberg 
+stores in sync across multiple operational systems. This made the analytics
```
Comment (mosabua) [inline _episodes/25.md:126]: As a result the analytics data is now much more dependable.

File: _episodes/25.md [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/25/zomato-after.png"/><br/>
+</p>
+
+## PR of the week: PR 4140 Implement aggregation pushdown in Pinot
+
+The [PR of the week](https://github.com/trinodb/trino/pull/6069) is actually a
+throwback to [episode thirteen](/episodes/13.html), _Trino takes a sip of Pinot_,
+where our guest [ElonAzoulay](https://twitter.com/ElonAzoulay) discussed some of
```
Comment (mosabua) [inline _episodes/25.md:137]: Elon Azoulay

File: _episodes/25.md [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+`COUNT(DISTINCT)` and `approx_distinct`. It is enabled by default and can be 
+disabled using the configuration property `pinot.aggregation-pushdown.enabled` 
+or the catalog session property `aggregation_pushdown_enabled`. 
+
+Thanks Elon!
+
+## Question of the week: Is there an array function that flattens a row like `1 | [a, b, c]` into three rows?
+
+Our [question of the week](https://trinodb.slack.com/archives/CFLB9AMBN/p1630241736052500)
+comes from Brian Hudson on the Slack channel. Brian is dealing with an `ARRAY`
```
Comment (mosabua) [inline _episodes/25.md:151]: on our Trino community on Slack.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: _episodes/25.md [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+throwback to [episode thirteen](/episodes/13.html), _Trino takes a sip of Pinot_,
+where our guest [ElonAzoulay](https://twitter.com/ElonAzoulay) discussed some of
+the upcoming features coming to the Pinot connector were. Push down aggregates
+was on that list and this just landed in the 361 release! 
+
+This PR implements aggregation pushdown for `COUNT`, `AVG`, `MIN`, `MAX`, `SUM`,
+`COUNT(DISTINCT)` and `approx_distinct`. It is enabled by default and can be 
+disabled using the configuration property `pinot.aggregation-pushdown.enabled` 
+or the catalog session property `aggregation_pushdown_enabled`. 
+
```
Comment (mosabua) [inline _episodes/25.md:145]: fyi https://github.com/trinodb/trino/pull/9208

File: _episodes/25.md [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+ * Lots of improvements for Hive view support
+ * Glue table statistics improvements
+ 
+More info at <https://trino.io/docs/current/release/release-361.html>.
+
+## Concept of the week: Change Data Capture
+
+If you know Trino, you know it allows for flexible architectures that include 
+many systems with varying use cases they support. We've come to accept this 
+potpourri of systems as a general modus operandi for most businesses.
```
Comment (mosabua) [inline _episodes/25.md:57]: remove usage of potpurri and modus operandi to make more understandable internationally

**Review by mosabua — APPROVED** [review]
Looks good now..

File: _episodes/25.md (hunk 17 above)
Comment (bitsondatadev) [reply _episodes/25.md:57]: yeah together it makes this a bit wordy. I'll remove potpourri.
